reconcile/aws_iam_password_reset.py

```diff
@@ -47,27 +47,35 @@ def run(dry_run, defer=None):
         reset_passwords = a.get("resetPasswords")
         if not reset_passwords:
             continue
-        with AWSApi(1, [a], settings=settings) as aws_api:
-            for r in reset_passwords:
-                user_name = r["user"]["org_username"]
-                request_id = r["requestId"]
-                state_key = f"{account_name}/{user_name}/{request_id}"
-                if state.exists(state_key):
-                    continue
-
-                role = get_roles(roles, user_name)
-                if not role:
-                    logging.error(f"Expected a role to be found with name {user_name}")
-                    sys.exit(1)
-
-                if not account_in_roles(role["roles"], account_name):
-                    logging.error(f"User {user_name} is not in account {account_name}")
-                    sys.exit(1)
-
-                logging.info(["reset_password", account_name, user_name])
-                if dry_run:
-                    continue
-
-                aws_api.reset_password(account_name, user_name)
-                aws_api.reset_mfa(account_name, user_name)
-                state.add(state_key)
+
+        aws_api: Optional[AWSApi] = None
+
+        for r in reset_passwords:
+            user_name = r["user"]["org_username"]
+            request_id = r["requestId"]
+            state_key = f"{account_name}/{user_name}/{request_id}"
+            if state.exists(state_key):
+                continue
+
+            role = get_roles(roles, user_name)
+            if not role:
+                logging.error(f"Expected a role to be found with name {user_name}")
+                sys.exit(1)
+
+            if not account_in_roles(role["roles"], account_name):
+                logging.error(f"User {user_name} is not in account {account_name}")
+                sys.exit(1)
+
+            logging.info(["reset_password", account_name, user_name])
+            if dry_run:
+                continue
+
+            if aws_api is None:
+                aws_api = AWSApi(1, [a], settings=settings)
+
+            aws_api.reset_password(account_name, user_name)
+            aws_api.reset_mfa(account_name, user_name)
+            state.add(state_key)
+
+        if aws_api:
+            aws_api.cleanup()
```

Comment:
this cleanup might not happen if the integration fails for whatever reason before it reaches this step

Reply:
Right, this should be fine, the cleanup is removing sessions. 
Nevertheless, I'll change it slightly to use context-managers again